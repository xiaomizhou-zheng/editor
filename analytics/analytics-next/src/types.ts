import { AnalyticsEventPayload } from './events/AnalyticsEvent';
import UIAnalyticsEvent from './events/UIAnalyticsEvent';

export type CreateUIAnalyticsEvent = (
  payload: AnalyticsEventPayload,
) => UIAnalyticsEvent;

export type AnalyticsEventCreator = (
  create: CreateUIAnalyticsEvent,
  props: Record<string, any>,
) => UIAnalyticsEvent | undefined;

export type CreateEventMapValue = AnalyticsEventPayload | AnalyticsEventCreator;

export type CreateEventMap = Record<string, CreateEventMapValue>;
