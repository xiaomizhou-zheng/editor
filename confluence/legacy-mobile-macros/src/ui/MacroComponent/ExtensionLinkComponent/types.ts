import { DocNode } from '@atlaskit/adf-schema';
import type { ExtensionParams } from '@atlaskit/editor-common/extensions';

export interface ExtensionLinkComponentProps {
  render: (document: DocNode) => JSX.Element;
  extension: ExtensionParams<any>;
}
