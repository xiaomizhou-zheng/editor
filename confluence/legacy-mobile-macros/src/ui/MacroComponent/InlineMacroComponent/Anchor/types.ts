import type { ExtensionParams } from '@atlaskit/editor-common/extensions';

export interface AnchorProps {
  extension: ExtensionParams<any>;
}
