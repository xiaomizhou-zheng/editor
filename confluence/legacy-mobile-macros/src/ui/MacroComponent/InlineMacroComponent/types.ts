import type { ExtensionParams } from '@atlaskit/editor-common/extensions';

export interface InlineMacroComponentProps {
  extension: ExtensionParams<any>;
}
