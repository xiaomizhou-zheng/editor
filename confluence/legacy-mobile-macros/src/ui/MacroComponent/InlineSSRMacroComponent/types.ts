import type { ExtensionParams } from '@atlaskit/editor-common/extensions';

export class CustomLegacyMacroError {
  constructor() {}
}

export interface CustomLegacyMacroResponse {
  html: string;
  resources: {
    css?: string;
    data?: string;
    js?: string;
  };
}

export interface InlineSSRMacroComponentProps {
  contentId: number;
  baseUrl: string;
  extension: ExtensionParams<any>;
  outputDeviceType: 'MOBILE' | 'DESKTOP';
  createPromise: Function;
  renderFallback: Function;
  onLinkClick: Function;
}
