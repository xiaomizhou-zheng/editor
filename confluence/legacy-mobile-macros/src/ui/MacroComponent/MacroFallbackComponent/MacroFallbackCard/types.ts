export type MacroCardType = {
  macroName: string;
  extensionKey: string;
  iconUrl?: any;
  action: any;
  errorMessage?: string;
  loading: boolean;
  secondaryAction: any;
};
