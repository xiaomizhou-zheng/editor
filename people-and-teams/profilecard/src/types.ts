import React from 'react';

import { IntlShape } from 'react-intl-next';

import {
  AnalyticsEventPayload,
  CreateUIAnalyticsEvent,
} from '@atlaskit/analytics-next';

import TeamCentralCardClient from './client/TeamCentralCardClient';
import TeamProfileCardClient from './client/TeamProfileCardClient';
import UserProfileCardClient from './client/UserProfileCardClient';

export interface ApiClientResponse {
  User: {
    id: string;
    isBot: boolean;
    isCurrentUser: boolean;
    avatarUrl: string | null;
    email: string | null;
    fullName: string | null;
    location: string | null;
    meta: string | null;
    nickname: string | null;
    companyName: string | null;
    remoteTimeString: string | null;
    remoteWeekdayIndex: string | null;
    remoteWeekdayString: string | null;
    status: StatusType;
    statusModifiedDate: number | null;
  };
}

export interface Team {
  // Header image props
  largeAvatarImageUrl?: string;
  smallAvatarImageUrl?: string;
  largeHeaderImageUrl?: string;
  smallHeaderImageUrl?: string;
  // Regular team details
  id: string;
  displayName: string;
  description: string;
  organizationId?: string;
  members?: {
    id: string;
    fullName: string;
    avatarUrl: string;
  }[];
}

export interface ProfileCardClientData {
  isBot: boolean;
  isCurrentUser: boolean;
  avatarUrl?: string;
  email?: string;
  fullName?: string;
  location?: string;
  meta?: string;
  nickname?: string;
  companyName?: string;
  timestring?: string;
  status: StatusType;
  statusModifiedDate?: number | null;
  customLozenges?: LozengeProps[];
}

export interface ReportingLinesUserPII {
  name: string;
  picture?: string;
}

export interface ReportingLinesUser {
  accountIdentifier: string;
  identifierType: 'ATLASSIAN_ID' | 'BASE64_HASH' | 'UNKNOWN';
  pii?: ReportingLinesUserPII;
}

export interface TeamCentralReportingLinesData {
  managers?: ReportingLinesUser[];
  reports?: ReportingLinesUser[];
}

export interface ProfileCardResourcedProps {
  userId: string;
  cloudId: string;
  resourceClient: ProfileClient;
  actions?: ProfileCardAction[];
  reportingLinesProfileUrl?: string;
  onReportingLinesClick?: (user: ReportingLinesUser) => void;
  position?: ProfilecardTriggerPosition;
  trigger?: TriggerType;
  children?: React.ReactNode;
  addFlag?: (flag: any) => void;
}

export interface ProfileCardResourcedState {
  visible?: boolean;
  isLoading?: boolean;
  hasError: boolean;
  error?: ProfileCardErrorType;
  data: ProfileCardClientData | null;
  reportingLinesData?: TeamCentralReportingLinesData;
  isKudosEnabled?: boolean;
  kudosDrawerOpen: boolean;
}

export interface ProfileCardTriggerProps {
  userId: string;
  /**
    A cloudId can be provided, and we'll verify that the target userId is an
    actual user in the specified site.

    Instead you can omit the cloudId and we won't do such a check.

    If you have a cloudId and only want to show users who are in that site
    then please provide it. If you're a site-less product or don't care about
    verifying that the shown user is in a particular site, don't provide a
    cloudId.
   */
  cloudId?: string;
  resourceClient: ProfileClient;
  actions?: ProfileCardAction[];
  reportingLinesProfileUrl?: string;
  onReportingLinesClick?: (user: ReportingLinesUser) => void;
  position?: ProfilecardTriggerPosition;
  trigger?: TriggerType;
  children?: React.ReactNode;
  testId?: string;
  addFlag?: (flag: any) => void;
}

export interface ProfileCardTriggerState {
  visible?: boolean;
  isLoading?: boolean;
  hasError: boolean;
  error?: ProfileCardErrorType;
  data: ProfileCardClientData | null;
  reportingLinesData?: TeamCentralReportingLinesData;
  shouldShowGiveKudos?: boolean;
  teamCentralBaseUrl?: string;
  kudosDrawerOpen: boolean;
}

export interface TeamProfileCardTriggerState {
  visible?: boolean;
  isLoading?: boolean;
  hasError: boolean;
  error?: any;
  data: Team | null;
  renderError?: boolean;
  shouldShowGiveKudos?: boolean;
  teamCentralBaseUrl?: string;
  kudosDrawerOpen: boolean;
}

export interface TeamProfilecardCoreProps {
  /**
    The id of the user viewing the profile card.

    This is used to determine whether to say that the member count is
    "including you" or not.
   */
  viewingUserId?: string;
  /**
    A list of extra buttons to be displayed at the bottom of the card.
    View Profile is always included by default.
   */
  actions?: ProfileCardAction[];
  /**
    A function allowing products to provide an href for the user avatars in the
    profilecard, e.g. so they can link to user's profile pages.
   */
  generateUserLink?: (userId: string) => string;
  /**
    A function allowing products to provide an onClick handler for when the
    user clicks on a user's avatar or avatar group item.
   */
  onUserClick?: (userId: string, event: React.MouseEvent<Element>) => void;
  /**
    This should be a link to the team's profile page. This will be used for:

    - Wrapping the trigger in a link to the team profile page (unless
      triggerLinkType is `none`).

    - Providing the link for the View Profile action button on the card.
   */
  viewProfileLink: string;
  /**
    An onClick action that navigates to the team's profile page. Something you
    may want, e.g. for an SPA site or tracking analytics of navigation. This
    is optional, just the viewProfileLink will suffice. Will be used for:

    - Adding an onClick to the trigger if the triggerLinkType is
      `clickable-link`.

    - Providing an onClick for the View Profile action button on the card.
   */
  viewProfileOnClick?: (event?: React.MouseEvent<Element>) => void;
}

export interface TeamProfileCardTriggerProps extends TeamProfilecardCoreProps {
  /** The id of the team. */
  teamId: string;
  /**
    The id of the organization that the team belongs to.
    Currently this is unused, but will become necessary in the future.
   */
  orgId: string;
  /** An instance of ProfileClient. */
  resourceClient: ProfileClient;
  /**
    The position relative to the trigger that the card should be displayed in.
   */
  position?: ProfilecardTriggerPosition;
  /**
    The interaction method used to trigger the team profile card to appear.

    - Click is generally recommended, but your needs may vary.

    - Hover works for mouse users, but does not support those who use a
      keyboard or screen reader, avoid using this if it's possible or makes
      sense.

    - Hover-click is usable for scenarios like inline-edits, where mouse users
      cannot click on the trigger without causing side effects, but keyboard
      users are still able to navigate into and trigger the profile card.

    Look at the "Team Profilecard Trigger" or "Trigger Link Types" examples to
    see how they behave, or ask in #team-twp-people-teams on Slack for our
    recommendations.
   */
  trigger?: 'hover' | 'click' | 'hover-click';
  /**
    We generally prefer to wrap the trigger in a link to the team profile
    page. This prop determines how that link behaves.

    - Link is generally the recommended prop (especially in combination with
      click or hover-click for the trigger prop above). It wraps the trigger in
      an anchor tag with the team profile link (that users can interact with
      via middle-click, etc.), but left clicking on the link is suppressed.

    - None does not wrap the trigger in a link at all. This makes it difficult
      for keyboard or screen reader users to know how to trigger the profile
      card. Generally avoid this.

    - Clickable-link wraps the trigger in a link with no special behaviour.
      This is suitable for places where you want the trigger to serve primarily
      as a link, and optionally allow hovering to preview the team first.

    Look at the example on "Trigger Link Types" for more in-depth analysis, or
    ask in #team-twp-people-teams on Slack for our recommendations.
   */
  triggerLinkType?: 'none' | 'link' | 'clickable-link';
  /**
    This is the component that will cause a team profile card to appear when
    interacted with according to the method specified by the trigger prop.
   */
  children?: React.ReactNode;
  /**
   * Used by the card to show Flags.
   */
  addFlag?: (flag: any) => void;
  /**
   * Optional cloudId. Pass this if rendering card within a sited context.
   */
  cloudId?: string;
}

export type StatusType = 'active' | 'inactive' | 'closed';

export type TriggerType = 'hover' | 'click';

export type StatusModifiedDateType =
  | 'noDate'
  | 'thisWeek'
  | 'thisMonth'
  | 'lastMonth'
  | 'aFewMonths'
  | 'severalMonths'
  | 'moreThanAYear';

export interface ProfileCardAction {
  callback?: (...args: any[]) => any;
  shouldRender?: (data: any) => boolean;
  id?: string;
  label: React.ReactNode;
  // Link to provide general link behaviour to the button. If both link and callback are provided link behaviour will be suppressed on click.
  link?: string;
}

export type LozengeColor =
  | 'default'
  | 'success'
  | 'removed'
  | 'inprogress'
  | 'new'
  | 'moved';

export interface LozengeProps {
  text: React.ReactNode;
  appearance?: LozengeColor; // defaults to 'default'
  isBold?: boolean; // defaults to false
}

export interface ProfilecardProps {
  isLoading?: boolean;
  hasError?: boolean;
  errorType?: ProfileCardErrorType;
  status?: StatusType;
  isBot?: boolean;
  avatarUrl?: string;
  fullName?: string;
  meta?: string;
  userId?: string;
  isCurrentUser?: boolean;
  // Nick name is also known as public name
  nickname?: string;
  email?: string;
  location?: string;
  companyName?: string;
  timestring?: string;
  actions?: ProfileCardAction[];
  clientFetchProfile?: () => void;
  statusModifiedDate?: number | null;
  withoutElevation?: boolean;
  /** Show manager and direct reports section on profile hover card, if available */
  reportingLines?: TeamCentralReportingLinesData;
  /** Base URL to populate href value for manager's and direct reports' user avatar  */
  reportingLinesProfileUrl?: string;
  /** Click handler when user clicks on manager's and direct reports' user avatar, un-clickable otherwise */
  onReportingLinesClick?: (user: ReportingLinesUser) => void;
  isKudosEnabled?: boolean;
  teamCentralBaseUrl?: string;
  addFlag?: (flag: any) => void;
  cloudId?: string;

  // Allow to pass custom message for disabled account which `status` prop is `inactive` or `closed`.
  // `disabledAccountMessage` should not contain react-intl-next components, ex: `FormattedMessage`,
  // because ProfileCard component is wrapped in its own `IntlProvider` and `FormattedMessage` will loads messages of `@atlaskit/profilecard`,
  // not from the consumer of `@atlaskit/profilecard`.
  disabledAccountMessage?: React.ReactNode;
  // Allow to show a status lozenge for disabled account which `status` prop is `inactive` or `closed`
  hasDisabledAccountLozenge?: boolean;
  // Allow consumers to pass in custom lozenges that will be displayed under the heading
  customLozenges?: LozengeProps[];
  openKudosDrawer?: () => void;
}

export type AnalyticsFromDuration = (duration: number) => AnalyticsEventPayload;

export type AnalyticsFunction = (generator: AnalyticsFromDuration) => void;

export interface AnalyticsProps {
  createAnalyticsEvent?: CreateUIAnalyticsEvent;
}

export interface AnalyticsWithDurationProps {
  fireAnalyticsWithDuration: AnalyticsFunction;
}

export interface TeamProfilecardProps extends TeamProfilecardCoreProps {
  /** Indicates whether the team's details are still loading. */
  isLoading?: boolean;
  /** Indicates whether an error occurred whilst fetching team details. */
  hasError?: boolean;
  /** Describes the type of error that occurred, if any. */
  errorType?: ProfileCardErrorType;
  /** The details of the team to be shown. */
  team?: Team;
  /** A callback that will try to re-fetch data in case an error occurred. */
  clientFetchProfile?: () => void;
  /** Details relevant to passing around analytics. */
  analytics: AnalyticsFunction;
}

export interface MessageIntlProviderProps {
  children: React.ReactNode;
  intl: IntlShape;
}

export type RelativeDateKeyType =
  | 'ThisWeek'
  | 'ThisMonth'
  | 'LastMonth'
  | 'AFewMonths'
  | 'SeveralMonths'
  | 'MoreThanAYear'
  | null;

export interface ProfileClient {
  flushCache: () => void;
  getProfile: (
    cloudId: string,
    userId: string,
    analytics?: (event: AnalyticsEventPayload) => void,
  ) => Promise<ProfileCardClientData>;
  getTeamProfile: (
    teamId: string,
    orgId?: string,
    fireAnalytics?: (event: AnalyticsEventPayload) => void,
  ) => Promise<Team>;
  getReportingLines: (userId: string) => Promise<TeamCentralReportingLinesData>;
  shouldShowGiveKudos: () => Promise<boolean>;
  getTeamCentralBaseUrl: () => string | undefined;
}

export type ProfilecardTriggerPosition =
  | 'bottom-start'
  | 'bottom'
  | 'bottom-end'
  | 'left-start'
  | 'left'
  | 'left-end'
  | 'top-end'
  | 'top'
  | 'top-start'
  | 'right-end'
  | 'right'
  | 'right-start';

export type ProfileCardErrorType = {
  reason: 'default' | 'NotFound';
} | null;

export interface ProfileClientOptions {
  url: string;
  gatewayGraphqlUrl?: string;
  cacheSize?: number;
  cacheMaxAge?: number;
  /** Enables Team Central functionality if enabled e.g. /gateway/api/watermelon/graphql*/
  teamCentralUrl?: string;
  /** URL to the Team Central app e.g. team.atlassian.com */
  teamCentralBaseUrl?: string;
  /** Name of integrating product e.g. jira, atlas, confluence **/
  productIdentifier?: string;
  cloudId?: string;
}

export interface ClientOverrides {
  userClient?: UserProfileCardClient;
  teamClient?: TeamProfileCardClient;
  teamCentralClient?: TeamCentralCardClient;
}
