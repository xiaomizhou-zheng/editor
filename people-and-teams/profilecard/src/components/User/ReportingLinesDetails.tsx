import React from 'react';

import { FormattedMessage } from 'react-intl-next';

import Avatar from '@atlaskit/avatar';
import AvatarGroup from '@atlaskit/avatar-group';
import Button from '@atlaskit/button/custom-theme-button';

import messages from '../../messages';
import {
  ManagerName,
  ManagerSection,
  OffsetWrapper,
  ReportingLinesHeading,
  ReportingLinesSection,
} from '../../styled/ReportingLines';
import {
  AnalyticsWithDurationProps,
  ProfilecardProps,
  ReportingLinesUser,
} from '../../types';
import { reportingLinesClicked } from '../../util/analytics';

export type ReportingLinesDetailsProps = Pick<
  ProfilecardProps,
  'reportingLines' | 'reportingLinesProfileUrl' | 'onReportingLinesClick'
> &
  AnalyticsWithDurationProps;

function getProfileHref(userId: string, profileUrl?: string) {
  return profileUrl ? profileUrl + userId : undefined;
}

const ReportingLinesDetails = (props: ReportingLinesDetailsProps) => {
  const {
    fireAnalyticsWithDuration,
    reportingLines = {},
    reportingLinesProfileUrl,
    onReportingLinesClick,
  } = props;
  const { managers = [], reports = [] } = reportingLines;

  const manager = managers.length >= 1 ? managers[0] : undefined;
  const hasReports = reports.length > 0;

  const getReportingLinesOnClick = (
    user: ReportingLinesUser,
    userType: 'manager' | 'direct-report',
  ) =>
    onReportingLinesClick
      ? () => {
          fireAnalyticsWithDuration((duration) =>
            reportingLinesClicked({
              duration,
              userType,
            }),
          );

          onReportingLinesClick(user);
        }
      : undefined;

  return (
    <>
      {manager && (
        <ReportingLinesSection>
          <ReportingLinesHeading style={{ marginBottom: 0 }}>
            <FormattedMessage {...messages.managerSectionHeading} />
          </ReportingLinesHeading>
          <OffsetWrapper>
            <Button
              appearance="subtle"
              spacing="none"
              href={getProfileHref(
                manager.accountIdentifier,
                reportingLinesProfileUrl,
              )}
              onClick={getReportingLinesOnClick(manager, 'manager')}
              isDisabled={!onReportingLinesClick}
            >
              <ManagerSection>
                <Avatar size="xsmall" src={manager.pii?.picture} />
                <ManagerName>{manager.pii?.name}</ManagerName>
              </ManagerSection>
            </Button>
          </OffsetWrapper>
        </ReportingLinesSection>
      )}
      {hasReports && (
        <ReportingLinesSection>
          <ReportingLinesHeading>
            <FormattedMessage {...messages.directReportsSectionHeading} />
          </ReportingLinesHeading>
          <AvatarGroup
            appearance="stack"
            size="small"
            data={reports.map((member: ReportingLinesUser, index) => {
              return {
                key: member.accountIdentifier,
                name: member.pii?.name || '',
                src: member.pii?.picture,
                href: getProfileHref(
                  member.accountIdentifier,
                  reportingLinesProfileUrl,
                ),
                onClick: getReportingLinesOnClick(member, 'direct-report'),
              };
            })}
            maxCount={5}
            testId="profilecard-reports-avatar-group"
          />
        </ReportingLinesSection>
      )}
    </>
  );
};

export default ReportingLinesDetails;
