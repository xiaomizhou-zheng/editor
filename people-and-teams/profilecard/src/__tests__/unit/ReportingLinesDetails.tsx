import React from 'react';

import { shallow } from 'enzyme';

import ReportingLinesDetails, {
  ReportingLinesDetailsProps,
} from '../../components/User/ReportingLinesDetails';
import { ReportingLinesUser } from '../../types';

describe('ReportingLinesDetails', () => {
  const exampleReportingLineUser: ReportingLinesUser = {
    accountIdentifier: 'abcd',
    identifierType: 'ATLASSIAN_ID',
    pii: {
      name: 'name',
      picture: 'picture',
    },
  };
  const defaultProps: ReportingLinesDetailsProps = {
    reportingLines: {
      managers: [exampleReportingLineUser],
      reports: [exampleReportingLineUser, exampleReportingLineUser],
    },
    reportingLinesProfileUrl: 'profile-url',
    fireAnalyticsWithDuration: () => {},
  };

  const renderShallow = (props = {}) =>
    shallow(<ReportingLinesDetails {...defaultProps} {...props} />);

  it('should match snapshot when manager and reports are specified', () => {
    expect(
      renderShallow({ onReportingLinesClick: () => null }),
    ).toMatchSnapshot();
  });

  it('should match snapshot when manager and reports are NOT specified', () => {
    expect(renderShallow({ reportingLines: {} })).toMatchSnapshot();
  });
});
