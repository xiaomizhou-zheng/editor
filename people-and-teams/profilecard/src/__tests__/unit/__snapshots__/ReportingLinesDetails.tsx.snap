// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportingLinesDetails should match snapshot when manager and reports are NOT specified 1`] = `<Fragment />`;

exports[`ReportingLinesDetails should match snapshot when manager and reports are specified 1`] = `
<Fragment>
  <styled.div>
    <styled.h5
      style={
        Object {
          "marginBottom": 0,
        }
      }
    >
      <MemoizedFormattedMessage
        defaultMessage="Manager"
        description="Title for a section on the profile card that show the user's direct manager"
        id="pt.team-profile-card.manager.heading"
      />
    </styled.h5>
    <styled.div>
      <CustomThemeButton
        appearance="subtle"
        href="profile-urlabcd"
        isDisabled={false}
        onClick={[Function]}
        spacing="none"
      >
        <styled.div>
          <Avatar
            size="xsmall"
            src="picture"
          />
          <styled.span>
            name
          </styled.span>
        </styled.div>
      </CustomThemeButton>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.h5>
      <MemoizedFormattedMessage
        defaultMessage="Direct reports"
        description="Title for a section on the profile card that show the user's direct reports"
        id="pt.team-profile-card.directReports.heading"
      />
    </styled.h5>
    <AvatarGroup
      appearance="stack"
      data={
        Array [
          Object {
            "href": "profile-urlabcd",
            "key": "abcd",
            "name": "name",
            "onClick": [Function],
            "src": "picture",
          },
          Object {
            "href": "profile-urlabcd",
            "key": "abcd",
            "name": "name",
            "onClick": [Function],
            "src": "picture",
          },
        ]
      }
      maxCount={5}
      size="small"
      testId="profilecard-reports-avatar-group"
    />
  </styled.div>
</Fragment>
`;
